rpi/controller.py

```diff
@@ -205,54 +205,36 @@ def run():
 
     while True:
         tag = b.get_tag()
-        m = Members()
-        roles = Roles()
-        member = m.get_by_tag(tag)
-        now = datetime.now()
-
-        log("Tag scanned: %s %s" % (tag, wiegandify(tag)))
-
-        if member == None:
-            log("Could not find member with tag %s." % (tag))
+        has_access, reason = test_auth(tag)
```

Comment:
A LOT of code has been pulled out of the main function. Can you talk about why this is needed?

Reply:
the code was similar (not quite the same, but the differences were mostly in how it was printed) to that of the test_auth function. I made it so that the main loop can be simplified to 3 clearer points. 
1: we get the fob from board.get_tag, which is harder to test without the physical setup running, and various fobs being tested.
2: we get if the fob scanned should be let in, and the reason that should be logged, through test_auth. Since we are using test_auth, we can test that the code works, without needing the physical setup.
3: unlocking the door/logging. I've left this mostly unchanged since I'm planning to update how we store logs.

The biggest improvement is that we can now test it. Before this, testing was done by running test_auth, with the code from the main loop pasted in there. But the comments there should be changed to clarify this